utils/frontend_handling.py

```diff
@@ -0,0 +1,159 @@
+    If the template can't be found, then we try again but using the default frontend.
+
+    What this ebhaviour achieves is that if a template is not found for a given frontend, then it is loaded from
+    the defualt frontend. This is similar to what could be ahieved using the `DIRS` parameter of the template
+    engines configuration but it is not exatcly the same. Using this wrapper, we make sure that templates from
+    the different frontend will never be mixed (e.g. we make sure that we don't take `sounds.html` from one
+    frontend and `base.html` from another).
+
+    This wrapper is intented to be use while we implement the new frontend. Once the whole implementation
+    process is finished and we don't need the two frontends to coexist anymore, then we can get rid of this
+    wrapper.
+    """
+
+    # Get the name of the template engine/frontend
+    name = selected_frontend(request)
+
+    try:
+        return django_render(request, template_name, context, content_type, status, using=name)
+    except TemplateDoesNotExist:
+
+        if name != settings.FRONTEND_DEFAULT:
+            # If the required template does can't be found using the dselected engine, try with the default engine
+            return django_render(request, template_name, context, content_type, status, using=settings.FRONTEND_DEFAULT)
+        else:
+            # If the default engine was being used, then raise the exception normally
+            raise
+
+
+def using_frontend(request, frontend_name):
+    """
+    Util functin to check which frontend is being used.
+    Returns True if `frontend_name` corresponds to the currently selecred frontend.
+    """
+    return selected_frontend(request) == frontend_name
+
+
+def using_beastwhoosh(request):
+    """
+    Returns True if currently used frontend is Beast Whoosh
+    """
+    return using_frontend(request, settings.FRONTEND_BEASTWHOOSH)
+
+
+def using_nightingale(request):
+    """
+    Returns True if currently used frontend is Nightingale
+    """
+    return using_frontend(request, settings.FRONTEND_NIGHTINGALE)
+
+
+def redirect_if_beastwhoosh(redirect_to_view):
+    """
+    Util decorator to be used in views which, when using Beast Whoosh frontend, should redirect to other views.
+    If there are pages in which the 2 front-end require the use of 2 completely different views, this decorator
+    can be used to point to the specific Beast Whoosh view. Use it as:
+
+    @redirect_if_beastwhoosh(new_view_func)
+    def old_view(request):
+        ...
+    """
+    def decorator(view_func):
+        def _wrapped_view(request, *args, **kwargs):
+            if not using_beastwhoosh(request):
+                return view_func(request, *args, **kwargs)
+            else:
+                return redirect_to_view(request, *args, **kwargs)
+        return _wrapped_view
+    return decorator
+
+
+def http_redirect_if_beastwhoosh(redirect_url_name, kwarg_keys=None, query_string=''):
```

Comment:
Ok, this makes it even worse 😄

Reply:
Then this one can be simply `redirect_redirect_if_beastwhoosh`